core/src/main/scala/io/qbeast/core/model/MetadataManager.scala

```diff
@@ -0,0 +1,56 @@
+package io.qbeast.core.model
+
+import io.qbeast.IISeq
+
+trait MetadataManager[DataSchema, FileDescriptor] {
+
+  /**
+   * Gets the Snapshot for a given table
+   * @param tableID the table identifier
+   * @return the current snapshot
+   */
+  def loadSnapshot(tableID: QTableID): QbeastSnapshot
+
+  /**
+   * Gets the Schema for a given table
+   * @param tableID the table identifier
+   * @return the current schema
+   */
+  def loadCurrentSchema(tableID: QTableID): DataSchema
+
+  /**
+   * Save methods
```

Comment:
Scaladoc format should not be used for naming parts of the source files, please use ordinary // or /*...*/.